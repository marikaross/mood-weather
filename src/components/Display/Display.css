.display {
  background-color: #fff;
  border: none;
  opacity: 0.8;
  width: 100%;
  height: 375px;
  margin: 0 auto;
  border-radius: 10px;
  padding-top: 5px;
  box-sizing: border-box;
}