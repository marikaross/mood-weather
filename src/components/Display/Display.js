import React from 'react';
import ReactAnimatedWeather from 'react-animated-weather';
import './Display.css';


const Display = ({conditions}) => {
    console.log(conditions)

  return(
    <div className="display"> 
       <ReactAnimatedWeather
        icon={conditions.icon}
        color={'#61dafb'}
        size={300}
        animate={true}
      />
      <h3>{conditions.temperature}&#8457;</h3>
      <h4>{conditions.summary}</h4>
    </div>

    )
}

export default Display;