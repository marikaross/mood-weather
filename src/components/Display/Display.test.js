import Display from './Display.js';
import React from 'react';
import ReactAnimatedWeather from 'react-animated-weather';
import { shallow } from 'enzyme';

describe('Display', () => {
  let wrapper, 
  mockConditions;

  beforeEach(() => {
    mockConditions = {icon: 'FOG', temperature: 1000, summary: 'Best day ever'}
    wrapper = shallow(<Display 
      conditions={mockConditions}/>);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });


});