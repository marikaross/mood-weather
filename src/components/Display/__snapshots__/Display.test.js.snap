// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Display
    conditions={
      Object {
        "icon": "FOG",
        "summary": "Best day ever",
        "temperature": 1000,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="conditions"
        />,
        <ReactAnimatedWeather
          animate={true}
          color="#61dafb"
          icon="FOG"
          size={275}
        />,
        <h3>
          1000
          ℉
        </h3>,
        <h4>
          Best day ever
        </h4>,
      ],
      "className": "display",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "conditions",
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "animate": true,
          "color": "#61dafb",
          "icon": "FOG",
          "size": 275,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            1000,
            "℉",
          ],
        },
        "ref": null,
        "rendered": Array [
          1000,
          "℉",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Best day ever",
        },
        "ref": null,
        "rendered": "Best day ever",
        "type": "h4",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="conditions"
          />,
          <ReactAnimatedWeather
            animate={true}
            color="#61dafb"
            icon="FOG"
            size={275}
          />,
          <h3>
            1000
            ℉
          </h3>,
          <h4>
            Best day ever
          </h4>,
        ],
        "className": "display",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "conditions",
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animate": true,
            "color": "#61dafb",
            "icon": "FOG",
            "size": 275,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              1000,
              "℉",
            ],
          },
          "ref": null,
          "rendered": Array [
            1000,
            "℉",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Best day ever",
          },
          "ref": null,
          "rendered": "Best day ever",
          "type": "h4",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
